src/modules/reachRandomJointPositions/src/module.cpp

```diff
@@ -238,129 +254,163 @@ bool reachRandomJointPositionsModule::configure(ResourceFinder &rf)
+            {
+                std::cerr << "GRID_MAPPING_WITH_RETURN mode available only for two controlled joints" << std::endl;
+                close_drivers();
+                return false;
+            }
+
+            center[0] = (controlledJoints[0].lower_limit+controlledJoints[0].upper_limit)/2;
+            center[1] = (controlledJoints[1].lower_limit+controlledJoints[1].upper_limit)/2;
+            semi_nr_of_lines[0] = ceil((controlledJoints[0].upper_limit-center[0])/controlledJoints[0].delta);
+            semi_nr_of_lines[1] = ceil((controlledJoints[1].upper_limit-center[1])/controlledJoints[1].delta);
+
+            //Start at the center of the workspace
+            listOfDesiredPositions.push_back(desiredPositions(center,return_point_waiting_period,
+                                                              true,desiredPositions::ROW_OUT));
+            bool return2center = (mode == GRID_MAPPING_WITH_RETURN);
+
+            //Draw rows moving second joint, for each step of first joint position
+            this->drawRow(center, 1, 0, return2center, false); //Draw center row
+            for(int i = 1; i < semi_nr_of_lines[0]+1; i++)
+            {
+                this->drawRow(center, 1, i, return2center, true); //Draw upper row
+
+                this->drawRow(center, 1, -i, return2center, false); //Draw lower row
+            }
+
+            //Return to center position for eventual drift checking by external module
+            listOfDesiredPositions.push_back(desiredPositions(center,return_point_waiting_period,
+                                                              true,desiredPositions::ROW_OUT));
+
+            //Draw rows moving first joint, for each step of second joint position
+            this->drawRow(center, 0, 0, return2center, false); //Draw center row
+            for(int j = 1; j < semi_nr_of_lines[1]+1; j++ )
+            {
+                this->drawRow(center, 0, j, return2center, true); //Draw upper row
+
+                this->drawRow(center, 0, -j, return2center, false); //Draw lower row
+            }
+
+            //Return to center position for eventual drift checking by external module
+            listOfDesiredPositions.push_back(desiredPositions(center,return_point_waiting_period,
+                                                              true,desiredPositions::ROW_OUT));
+
+            //Print list of desired positions
+            std::cout  << "List of desired positions: " << std::endl;
+            for(int i=0; i<int(listOfDesiredPositions.size()); i++)
+            {
+                std::cout << listOfDesiredPositions[i].toString() << std::endl;
+            }
+
+            isTheRobotInReturnPoint.open("/"+moduleName+"/"+controlledJoints[0].part_name+"/isTheRobotInReturnPoint:o");
+            useFurtherPosForFitting.open("/"+moduleName+"/"+controlledJoints[0].part_name+"/useFurtherPosForFitting:o");
+            
+            break;
```

Comment:
yes, it works outside the scope. I've checked we actually "break" after this line.